src/stan_math_backend/Stan_math_code_gen.ml

```diff
@@ -176,11 +176,21 @@ let pp_method ppf rt name params intro ?(outro = nop)
   outro ppf () ;
   pf ppf "@,} // %s() @,@]" name
 
+let pp_extend_vector name type_ pp_elt ppf elts =
+  if List.is_empty elts then nop ppf elts
+  else
+    pf ppf
+      "@[<hov 2>%s temp@ {%a};@]@,\
+       %s.reserve(%s.size() + temp.size());@,\
+       %s.insert(%s.end(), temp.begin(), temp.end());@,"
+      type_ (list ~sep:comma pp_elt) elts name name name name
+
 (** Print the [get_param_names] method of the model class
   @param ppf A pretty printer.
  *)
 let pp_get_param_names ppf {Program.output_vars; _} =
   let add_param = fmt "%S" in
+  (* old -- kept as overload for compat until model_base changes *)
```

Comment:
del this and the other new comment tho'

Reply:
After the change to model_base I plan on deleting the old code, and those comments

Reply:
Lets not delete the old code since we want to keep backwards compatibility (C++ sees these as two different virtual function signatures)

Reply:
We cannot expose both 

`  virtual void get_param_names(std::vector<std::string>& names) const = 0;` 

and 

`  virtual void get_param_names(std::vector<std::string>& names,  const bool emit_transformed_parameters = true, const bool emit_generated_quantities = true) const = 0;`

since it makes calls with only the vector argument ambiguous. Because we're providing default values for the arguments which match previous behavior, we should just keep the second